x-pack/plugins/enterprise_search/server/lib/indices/utils/get_index_data.ts

```diff
@@ -84,3 +88,41 @@ export const getIndexDataMapper = (totalIndexData: TotalIndexData) => {
       indexName
     );
 };
+
+function isHidden(index: IndicesIndexState): boolean {
+  return Boolean(index.settings?.index?.hidden) || index.settings?.index?.hidden === 'true';
```

Comment:
I like that we moved this a guard, but this would return `true` if `index.settings?.index?.hidden === 'false'`
```
Welcome to Node.js v16.19.1.
Type ".help" for more information.
> Boolean('true')
true
> Boolean('false')
true
> 
```

Granted I think the actual possible states of this are `'true'` or `undefined`

Reply:
🤦 
This can actually be 'false', if an index has been set to hidden and then moved to not-hidden. I'll create a PR to fix this.